src/tpm2_pytss/ESAPI.py

```diff
@@ -2179,6 +2179,68 @@ def hmac(
         )
         return TPM2B_DIGEST(_get_dptr(out_hMAC, lib.Esys_Free))
 
+    def mac(
+        self,
+        handle: ESYS_TR,
+        buffer: Union[TPM2B_MAX_BUFFER, bytes, str],
+        hash_alg: TPM2_ALG,
+        session1: ESYS_TR = ESYS_TR.PASSWORD,
+        session2: ESYS_TR = ESYS_TR.NONE,
+        session3: ESYS_TR = ESYS_TR.NONE,
+    ) -> TPM2B_DIGEST:
+        """Invoke the TPM2_MAC command.
+
+        This function invokes the TPM2_MAC command in a one-call
+        variant. This means the function will block until the TPM response is
+        available.
+
+        Args:
+            handle (ESYS_TR): Handle for the symmetric signing key providing the HMAC key.
+            buffer (Union[TPM2B_MAX_BUFFER, bytes, str]): MAC data.
+            hash_alg (TPM2_ALG): Algorithm to use for MAC.
+            session1 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.PASSWORD.
+            session2 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.NONE.
+            session3 (ESYS_TR): A session for securing the TPM command (optional). Defaults to ESYS_TR.NONE.
+
+        Raises:
+            TypeError: If a parameter is not of an expected type.
+            ValueError: If a parameter is not of an expected value.
+            TSS2_Exception: Any of the various TSS2_RC's the lower layers can return.
+
+        Returns:
+            A TPM2B_DIGEST result of the MAC.
+
+        C Function: Esys_MAC
+
+        TPM Command: TPM2_MAC
+        """
+        if not _lib_version_atleast("tss2-esapi", "4.0.0"):
+            raise NotImplementedError("ESAPI v<4.0.0 does not support MAC Interface")
+
+        _check_handle_type(handle, "handle")
+        _check_handle_type(session1, "session1")
+        _check_handle_type(session2, "session2")
+        _check_handle_type(session3, "session3")
+
+        _check_friendly_int(hash_alg, "hash_alg", TPM2_ALG)
+
+        buffer_cdata = _get_cdata(buffer, TPM2B_MAX_BUFFER, "buffer")
+
+        out_hMAC = ffi.new("TPM2B_DIGEST **")
```

Comment:
out_HMAC, I think this needs to be out_MAC to work. My question is how are the tests passing :-p

Reply:
I think thats because the first test is to check for v >= 4.0 and we dont have a 4.0 job so it never goes past that. I'll fix the out_HMAC